daemon.go

```diff
@@ -14,156 +14,53 @@
 package main
 
 import (
-	"time"
-
-	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 // Config section
 
-// daemon configuration
-type daemonConfig struct {
-	serverpollInterval time.Duration
-	serverURL          string
-	deviceID           string
-}
-
-// Daemon section
-
 type menderDaemon struct {
-	Updater
-	UInstallCommitRebooter
-	Controler
-	config      daemonConfig
+	mender      Controler
 	stopChannel chan (bool)
 	stop        bool
 }
 
-func NewDaemon(client Updater, device UInstallCommitRebooter,
-	mender Controler) *menderDaemon {
-
-	config := mender.GetDaemonConfig()
+func NewDaemon(mender Controler) *menderDaemon {
 
 	daemon := menderDaemon{
-		Updater:                client,
-		UInstallCommitRebooter: device,
-		Controler:              mender,
-		config:                 config,
-		stopChannel:            make(chan bool),
+		mender:      mender,
+		stopChannel: make(chan bool),
 	}
 	return &daemon
 }
 
-func (daemon menderDaemon) StopDaemon() {
-	daemon.stopChannel <- true
+func (d menderDaemon) StopDaemon() {
```

Comment:
I think that the original intention was to be able to react to SIGINT/SIGTERM, but I don't see `StopDaemon()` being called anywhere.